main.go

```diff
@@ -150,6 +149,10 @@ func buildRequestHandler(watchdogConfig config.WatchdogConfig) http.HandlerFunc
 		break
 	}
 
+	if watchdogConfig.MaxInflight > 0 {
```

Comment:
technically, the if block isn't needed because the factory already handles the `<=0` case